### ISSUE TITLE
Aeon: enabling request links on library/locations and routing requests to Aeon

### ISSUE THREAD

**Issue by saseestone** [sul-dlss/sul-requests#1333]
With the implementation of Aeon, we need to:
- configure currently enabled library or library/location request links to go to Aeon instead of Symphony
- enable some new library/location/item type combinations for request links, and direct those requests to Aeon

For each row below, we should be directing the request link to Aeon **EXCEPT** when a Finding Aid is present in the MARC 856. If there is a finding aid in the 856, we need to route the request to the corresponding collection guide specified in the 856. (We do this currently for library: HV-ARCHIVE and HOOVER requests.)

**Currently enabled mediated paging library, and all requests need to go to Aeon**
 - library: RUMSEYMAP
 - library: SPEC-COLL

> Note that we should continue to suppress request links for SPEC-COLL items that have a current location of INPROCESS, MISSING, ON-ORDER, SPEC-INPRO, or ENDPROCESS. https://github.com/sul-dlss/SearchWorks/blob/60379f305e1cbfbf2cf98ff4a45f36c533be2238/config/settings.yml#L559

**Currently enabled library/location combinations, and requests need to go to Aeon**
- library: SAL, location: L-PAGE-EA, Item type = NH-INHOUSE
- library: SAL3, location: PAGE-AS, Item type = NH-INHOUSE
- library: SAL3, location: PAGE-EA, Item type = LCKSTK

> Note SAL3 PAGE-EA items that do NOT have item type = LCKSTK should display request links, but the request should go to Symphony.
 

**NEW library/location/item type combinations that need to be enabled for requests, and then requests go to Aeon**
- library: ARS, location: RECORDINGS, item type: NH-RCORDNG
- library: EAST-ASIA, location: LOCKED-STK, Item type = LCKSTK
- library: EAST-ASIA, location: LOCK-CHN, Item type = LCKSTK
- library: EAST-ASIA, location: LOCK-STK-O, item type = LCKSTK

**Comment by saseestone** [comment]
@ggeisler will be producing related tickets to describe the userflow from SW to an interstitial page in sul-requests to Aeon. He's also produced wireframes to illustrate the flow.

**Comment by ggeisler** [comment]
At a general level, there are three distinct use cases by which a user goes from Searchworks to Aeon:

- Single volume/copy item
- Multi-volume/copy item
- Finding aid item

All three cases should display a Requests informational page after the user selects the request button. The content of the Requests informational page differs slightly based on which of the three use cases is involved, and which library/location owns the selected item. To keep this ticket from being too complicated, there's a separate ticket for the [Requests informational page details](https://github.com/sul-dlss/sul-requests/issues/1335).

In terms of page flow, the three cases differ a bit in what comes after the user selects "Continue" on the Requests informational page.

---

<img width="1034" alt="SW-to-Aeon-use-case-single-vol" src="https://user-images.githubusercontent.com/101482/188241154-5ab41b4c-5ad8-4d0f-a15c-8878baa79b87.png">

This is the simplest case. After selecting "Continue" on the Requests informational page, the flow should pass to Aeon, where the user logs in (or registers and sees the first time info page, if necessary) and then goes to the Aeon New request page to complete details for requesting the item.

Here are Figma clickable prototypes that show the page flow for the single volume/copy use case. Each is basically the same page flow, but the variations show how the text on the Requests informational page varies slightly by library/location:

- [SPEC](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1008%3A176&scaling=scale-down&page-id=1008%3A175&starting-point-node-id=1008%3A176)
- [EAL](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1134%3A1103&scaling=scale-down&page-id=1134%3A1101&starting-point-node-id=1134%3A1103)
- [ARS](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1120%3A747&scaling=scale-down&page-id=1120%3A745&starting-point-node-id=1120%3A747)
- [Rumsey](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1134%3A1252&scaling=scale-down&page-id=1134%3A1250&starting-point-node-id=1134%3A1252)

---

<img width="1034" alt="SW-to-Aeon-use-case-multi-vol" src="https://user-images.githubusercontent.com/101482/188241264-5b98c79e-f8b3-42a1-bf19-2291cd52ff72.png">

Because this use case involves more than one requestable item (volume/copy), there is an additional selection page in Requests for the user to indicate which item(s) they want to request. This should be similar to the existing page, but simpler because we need to remove the appointment and comment sections, which are now handled by Aeon. See the ticket about [updating the Requests selection page](https://github.com/sul-dlss/sul-requests/issues/1336) for details.

Here's a Figma clickable prototype that shows the page flow for the multi-volume use case: [SPEC](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1009%3A238&scaling=scale-down&page-id=1009%3A14&starting-point-node-id=1009%3A23)

---

<img width="1034" alt="SW-to-Aeon-use-case-finding-aid" src="https://user-images.githubusercontent.com/101482/188241274-375fc932-5fba-4779-85ed-00100b1819eb.png">

For a finding aid item, the page flow includes OAC. Unlike the two cases above, when a user selects the "Continue" button on the Requests informational page, they should be redirected to the OAC page for the selected finding aid collection. The "Request items" button in OAC will pass the user onto Aeon.

Here are Figma clickable prototypes that show the page flow for the finding aid use case. There's a variation for SPEC, EAL, and ARS (Rumsey doesn't have a finding aid item). The page flow for the SPEC finding aid example assumes you need to logon to Aeon for the first time, so includes all of the Aeon pages for first-time visitors, registration, etc. 

(All the other prototypes I've linked to, besides the SPEC finding aid one, assume the user has already registered with Aeon and is logging in via SSO, so they go straight from the Continue button on the Requests informational page to Aeon (or for the EAL and ARS finding aid examples, straight from OAC to Aeon)).

- [SPEC](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1022%3A237&scaling=scale-down&page-id=1008%3A21&starting-point-node-id=1022%3A146)
- [EAL](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1134%3A976&scaling=scale-down&page-id=1134%3A975&starting-point-node-id=1134%3A976)
- [ARS](https://www.figma.com/proto/CquVTLOGUVavYkwkUWs0VQ/SW---Requests---Aeon?node-id=1119%3A621&scaling=scale-down&page-id=1119%3A614&starting-point-node-id=1119%3A621)

**Comment by thatbudakguy** [comment]
question for @saseestone: @cbeer and I found some ARS items with a RECORDINGS location that don't have the type NH-RCORDNG. this seems a little suspicious because there are ~50,000 overall items in RECORDINGS but only ~75 without the NH-RCORDNG type. an example is https://searchworks.stanford.edu/view/9266429, which instead has type NH-DVDCD. is this OK?

**Comment by saseestone** [comment]
Yes, there are ARS materials with Location = RECORDINGS and an Item Type other than NH-RCORDNG. Much of it is dirty data, but not all of it.  Staff specifically asked to only place request links on materials with Item Type = NH-RCORDNG.

**Comment by saseestone** [comment]
ARS is in the process of changing their metadata practice so that Library ARS, Location = RECORDINGS should always result in Item Type = NH-RCORDNG.  So they are getting to a place where Library/Location is all that will be needed.  

They have confirmed that they can manage if we place the request links at the Library/Location level, and disregard the Item Type distinction.

**Comment by thatbudakguy** [comment]
ok — so, the new rule for items that should go to aeon is:
* library: ARS, location: RECORDINGS 

irrespective of item type. thanks!
